core/src/serve_repair.rs

```diff
@@ -222,6 +222,17 @@ impl RepairRequestHeader {
             nonce,
         }
     }
+
+    #[cfg(test)]
+    pub fn default_for_tests() -> Self {
```

Comment:
Would be much better to use the actual api to construct a valid one.
This one is just like  adding a backdoor.
At the very least, this does not need to be `pub` and can be inside `mod tests`.

Reply:
not sure how it's a backdoor with a `cfg(test)` annotation, but moved to mod test.